spec/services/metadata/handlers/teacher_spec.rb

```diff
@@ -0,0 +1,202 @@
+RSpec.describe Metadata::Handlers::Teacher do
```

Comment:
can we add specs for `nil` values, or when someone is an ect and not a mentor, or a teacher without any training periods etc.